File: php/classes/SHA256.php
<?php

class SHA256
{
	/**
	* Computes the SHA-256 checksum.
	*/
	public static function compute($text)
	{
		return hash("sha256", $text, true);
	}
	
	/**
	* Computes the HMAC for SHA-256.
	*/
	public static function computeHMAC($key, $text)
	{
		return hash_hmac("sha256", $text, $key, true);
	}
	
}

?>

File: csharp/ASCrypt/SHA256.cs
using System;
using System.Security.Cryptography;

namespace ASCrypt
{
    public class SHA256
    {
        /// <summary>
        /// Computes the SHA-256 checksum for the bytes.
        /// </summary>
        public static Byte[] Compute(Byte[] bytes)
        {
            SHA256Managed sha256 = new SHA256Managed();
            return sha256.ComputeHash(bytes);
        }

        /// <summary>
        /// Computes the HMAC-SHA-256 for the key and bytes.
        /// </summary>
        public static Byte[] ComputeHMAC(Byte[] key, Byte[] bytes)
        {
            HMACSHA256 hmac = new HMACSHA256(key);
            return hmac.ComputeHash(bytes);
        }

    }

}
